File: StaffAdministrationCsharp/StaffAdministrationCsharp/Boss.cs
namespace StaffAdministrationCsharp;

public class Boss : Manager
{
    public decimal StockOptions;

    public Boss(string name, decimal yearlySalary, Department department, decimal bonus, decimal stockOptions) : base(name, yearlySalary, department,  bonus)
    {
        StockOptions = stockOptions;
        CompanyDepartment(true);
    }

    public decimal CalculateMonthlySalary()
    {
        decimal monthlySalary = (YearlySalary + Bonus + StockOptions) / 12;
        return monthlySalary;
    }
    
    public void DisplayInformation()
    {
        Console.WriteLine($"Name: {Name}, Monthly salary: {CalculateMonthlySalary()}, Department: {Department}");
    }
    
    public void CompanyDepartment(bool boss)
    {
        if (boss) { Department = Department.Company; }
    }
}

File: StaffAdministrationCPP/Boss.cpp
//
// Created by stepa on 1/20/2025.
//

#include "Boss.h"
#include <iostream>

Boss::Boss(string name, float yearly_salary, department department, float bonus, float stockOptions) : Manager(name, yearly_salary, department, bonus)
{
 StockOptions = stockOptions;
}

float Boss::CalculateMonthlySalary()
{
    float monthly_salary = (YearlySalary + Bonus + StockOptions) / 12 ;
    return monthly_salary;
}

void Boss::DisplayInformation()
{

    cout << "Name: " << Name << " Monthly salary: " << CalculateMonthlySalary() << " Department: " << Department << endl;
}